### ISSUE TITLE
Filterable Nav - use fully qualified node path when comparing identity. 

### ISSUE THREAD

**Issue by EvanBoyle** [pulumi/theme#87]
Follow up from https://github.com/pulumi/theme/pull/84

https://github.com/pulumi/theme/pull/84#pullrequestreview-909394966

We made a fix that added the root name to the identity check. We should use the full node name that is computed earlier in the function:

https://github.com/pulumi/theme/blob/master/stencil/src/components/pulumi-api-doc-filterable-nav/pulumi-api-doc-filterable-nav.tsx#L97

This identifier should be unique, where as just comparing by name+root will have the same problems for a tree structure such as:

`foo.v1.bar.v1.buzz`

`foo.v1` and `foo.v1.bar.v1` would have the same equality if you just compare the root and leaf names rather than the fully qualified path.

**Comment by susanev** [comment]
@kimberleyamackenzie im going to add this into the next iteration. let me know if you think you might be able to do it sooner.

**Comment by kimberleyamackenzie** [comment]
I will look at this soon - when I made the original fix, I tried to used the fully qualified path, but there were actually some road bumps.  I'll give this a second look during this iteration, and if it's quick, I'll do it; otherwise I'll update this issue with some specifics about why and tackle during 71.  So I'll stick it in this one for now.

**Comment by kimberleyamackenzie** [comment]
Also @EvanBoyle just to make sure I'm understanding your example correctly

> foo.v1.bar.v1.buzz
> 
> foo.v1 and foo.v1.bar.v1 would have the same equality if you just compare the root and leaf names rather than the fully qualified path.

In the case of `foo.v1.bar.v1`, this would be a tree depth of 3, is that right? `foo` as the node, and `v1`, `bar`, and `v1` each as descending children?  If I'm reading that correctly, I just want to clarify that I believe the tree was explicitly built to only handle a depth of 2 (`foo.v1.bar`), and it will not function accurately for additional depths as it currently works.  That doesn't take away from tackling this issue to use the fully qualified path, but I just wanted to make sure we're on the same page about that point.

**Comment by EvanBoyle** [comment]
> In the case of foo.v1.bar.v1, this would be a tree depth of 3, is that right?

yes. But I think this would also reproduce with something like `foo.v1.foo`. 

> the tree was explicitly built to only handle a depth of 2

I was not aware of this. I don't believe our provider schema has this limitation, and that we could see arbitrary node depths.
